### ISSUE TITLE
Upgrade to newer version of stylelint 

### ISSUE THREAD

**Issue by ludovig** [olegskl/gulp-stylelint#132]
Hello,

I had some trouble getting gulp-stylelint working. As a config is needed, I installed stylelint-config-recommanded, 
`npm install stylelint-config-recommended --save-dev`
But newer version 4.0.0 of this package throw this error : 
'Undefined rule declaration-block-no-duplicate-custom-properties' (as it's only compatible with 13.12 version of stylelint)
And previous version 3.0.0 doesn't work either, with this error : 
'Undefined rule function-calc-no-invalid' (as it only compatible with stylelint version > 10.1.0)
So I had to switch back to version 2.2.0 that is compatible with stylelint ^8.0 version provided by gulp-stylelint.

Could you upgrade from 8.0.0 that was released in mid-2017 to latest version that was released on 6 march of this year ?

**Comment by danny007in** [comment]
u needed to upgrade stylelint plugin

**Comment by Piedone** [comment]
Did you see this @olegskl? Currently, stylelint v15 can't be used with this plugin.

**Comment by rollecode** [comment]
Can this be done @olegskl?

**Comment by rollecode** [comment]
I have sent a PR #138 but in case nothing happens my fork works fine for itself. For someone else who relies on gulp-stylelint in their daily work like me, please install my fork to your project, I try to keep its stylelint and dependencies up to date:

```shell
npm install @ronilaukkarinen/gulp-stylelint --save-dev
```

Tested & working just now.

**Comment by Piedone** [comment]
Thank you, Roni!

BTW force installing gulp-stylelint with stylelint v15 works, didn't notice any issues.

**Comment by rollecode** [comment]
Yeah it works if you are forcing them inside gulp-stylelint but if you want to use the stable tag in projects without going through the node module level you have to do the installs separately.

**Comment by pensierinmusica** [comment]
@olegskl any update on this?

**Comment by rollecode** [comment]
> BTW force installing gulp-stylelint with stylelint v15 works, didn't notice any issues.

Yeah, but force-installing is always a bad practice as it may lead to other unexpected errors.

**Comment by jds-tag** [comment]
Thank you @ronilaukkarinen !! https://github.com/olegskl/gulp-stylelint/issues/132#issuecomment-957267876 works great. But an official release would be nice for everyone. @olegskl can you please release a v14? I see #138 is merged so we should be pretty close, right? High-fives to the Open Source community 🤚 ☮️

**Comment by Gomes81** [comment]
Is there new version planned soon? Would be great to be able to upgrade to stylelint 14 and afterwards.
Many thanks for the great job done here.

**Comment by rollecode** [comment]
@Gomes81 No, it is not as we have not heard the maintainer @olegskl for almost 3 years now. My fork still works, see [the comment earlier on this issue](https://github.com/olegskl/gulp-stylelint/issues/132#issuecomment-957267876).

**Comment by JacobDB** [comment]
Too bad @olegskl has seemingly abandoned this project 🙄 this doesn't seem like it'd be that hard to fix, especially given there's a PR...

@ronilaukkarinen, any chance you can update your fork? Looks like it's not compatible with stylelint v16

**Comment by rollecode** [comment]
Yes, there is a pending PR, but I'm still using v15 for stylelint plugins that depend on gulp-stylelint v16 and haven't been able to get migrated those yet. Scheduled for January 2024 with our team, I'll get back to this as soon as I can. Ref: [#10](https://github.com/developer-stylechain/gulp-stylelint/issues/10).

**Comment by xavierbourretsicotte** [comment]
Maybe @XhmikosR  can help ?

**Comment by xavierbourretsicotte** [comment]
Hey @olegskl  where are you ?
